nixos/modules/services/torrent/rtorrent.nix

```diff
@@ -0,0 +1,208 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.rtorrent;
+
+in {
+  options.services.rtorrent = {
+    enable = mkEnableOption "rtorrent";
+
+    dataDir = mkOption {
+      type = types.str;
+      default = "/var/lib/rtorrent";
+      description = ''
+        The directory where rtorrent stores its data files.
+      '';
+    };
+
+    downloadDir = mkOption {
+      type = types.str;
+      default = "${cfg.dataDir}/download";
+      description = ''
+        Where to put downloaded files.
+      '';
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "rtorrent";
+      description = ''
+        User account under which rtorrent runs.
+      '';
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "rtorrent";
+      description = ''
+        Group under which rtorrent runs.
+      '';
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.rtorrent;
+      defaultText = "pkgs.rtorrent";
+      description = ''
+        The rtorrent package to use.
+      '';
+    };
+
+    port = mkOption {
+      type = types.int;
+      default = 50000;
+      description = ''
+        The rtorrent port.
+      '';
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to open the firewall for the port in <option>services.rtorrent.port</option>.
+      '';
+    };
+
+    rpcSocket = mkOption {
```

Comment:
Does this need to be an option? If this value is changed it will probably break the service. If this needs to be an option I suggest documenting the details behind it (such as the sysadmin needs to provision the directory the socket is in if not `/run/rtorrent`).

Reply:
The idea was to access its value from another module since I'm also writing a `rutorrent` module (a web frontend for rtorrent). But there is probably a way to have an attribute accessible while not being an option, is there ? Or I could even hard-code it on both ends. The only use case for a different unix socket path would be to run multiple instances but I don't think that's possible given the way this module is made.

I was also thinking about adding a way to configure rpc through tcp socket. You can configure rtorrent's rpc either way but since it doesn't provide authentification you always end up adding auth with nginx as a proxy. I don't see why someone would want to run rtorrent and nginx on different servers or containers so I think a unix socket is enough. But maybe I shouldn't be making such assumptions and provide mutually exclusive options for both.

Reply:
You can do `readOnly = true` to not make it assignable, seems like a good use-case for this

Reply:
Thanks, I think this should be enough unless someone has a use case for rpc through tcp socket.